backends/bmv2/jsonconverter.cpp

```diff
@@ -803,13 +805,34 @@ JsonConverter::JsonConverter(const CompilerOptions& options) :
         headerParameter(nullptr), userMetadataParameter(nullptr), stdMetadataParameter(nullptr)
 {}
 
+void addSrcInfoData(Util::JsonObject* result,
```

Comment:
Hi Andy,
Here are my thoughts -- not quite at the level of CPR though :) Note that in order to be a cleaner implementation, this might require a bit more work on your part, but I believe that in the long run it will adhere better to the design that we have:
1. The JSONConverter should really be a visitor pass. It has not been implemented as such, but @hanw is working on refactoring it so that it conforms to the rest of our passes.
2. Each IR node should know its source information (or know that it doesn't have it). We should define an interface that will be "mixed-in" (or even declare these methods in the IR::Node itself) so that when a node is visited it can output the source info -- just as you described before with the Node::addSourceInfoToJson(Util::JsonObject *jsonObj) function.
3. You can pick and chose which nodes output the info by adding it to the visitor functions.

The reason I'm proposing this is that you will visit each node in the JSONConverter visitor and moreover, outputing source info is not separable into another visitor (unless you can somehow re-traverse the whole JSON tree). The addSrcInfoData functions that you wrote would work well for such a separate visitor. By encapsulating the source info into the IR node, you can also carry it through transforms (e.g., if you inline a table and create clone nodes, both sites should refer to the original source).

Let me know what you think.